lib/puppet_x/puppetlabs/scheduled_task/trigger.rb

```diff
@@ -389,42 +390,44 @@ def self.bitmask_to_names(bitmask)
 
       # Defines the day of the month the task will run.
       class Days
-        # 32 bits for 31 possible days to set + value 'last'
-        MAX_VALUE = 0b11111111111111111111111111111111
+        # 32 bit mask, but only 31 days can be set.
+        # This is contrary to the V2 IMonthlyTrigger::DaysOfMonth documentation
+        # referenced below, but in testing, setting the last bit of the bit
+        # mask does not set 'last' day of month as that documentation suggests.
+        # Instead it results in an error. That feature will be handled instead
+        # by the RunOnLastDayOfMonth property of the trigger object.
+        # V2 IMonthlyTrigger::RunOnLastDayOfMonth
+        # https://docs.microsoft.com/en-us/windows/win32/api/taskschd/nf-taskschd-imonthlytrigger-put_runonlastdayofmonth
+        MAX_VALUE = 0b01111111111111111111111111111111
 
         # V1 MONTHLYDATE structure
         # https://msdn.microsoft.com/en-us/library/windows/desktop/aa381918(v=vs.85).aspx
         # V2 IMonthlyTrigger::DaysOfMonth
         # https://msdn.microsoft.com/en-us/library/windows/desktop/aa380735(v=vs.85).aspx
         def self.indexes_to_bitmask(day_indexes)
-          day_indexes = [day_indexes].flatten.map do |m|
-            # The special "day" of 'last' is represented by day "number"
-            # 32. 'last' has the special meaning of "the last day of the
-            # month", no matter how many days there are in the month.
-            # raises if unable to convert
-            (m.is_a?(String) && m.casecmp('last').zero?) ? 32 : Integer(m)
+          integer_days = [day_indexes].flatten.select { |i| i.is_a?(Integer) }
+          invalid_days = integer_days.reject { |i| i.between?(1, 31) }
+          if day_indexes.nil? || (day_indexes.is_a?(Hash) && day_indexes.empty?)
```

Comment:
Should we move this check up to the start of the method on line 408, before we do anything else with `day_indexes`?

Reply:
Yup. We can do that.